Sources/SwiftDocCUtilities/ArgumentParsing/Subcommands/DiffDocCArchive.swift

```diff
@@ -0,0 +1,253 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import ArgumentParser
+import Foundation
+import SwiftDocC
+
+extension Docc.ProcessArchive {
+    
+    struct DiffDocCArchive: ParsableCommand {
+        
+        // MARK: - Content and Configuration
+        
+        /// Command line configuration.
+        static var configuration = CommandConfiguration(
+            commandName: "diff-docc-archive",
+            abstract: "Produce a markdown file saved as {FrameworkName}_ChangeLog.md containing the diff of added/removed symbols between the two provided DocC archives.",
+            shouldDisplay: true)
+        
+        /// Content of the 'changeLog' template.
+        static func changeLogTemplateFileContent(
+            frameworkName: String,
+            initialDocCArchiveVersion: String,
+            newerDocCArchiveVersion: String,
+            additionLinks: String,
+            removalLinks: String
+        ) -> [String : String] {
+            [
+                "\(frameworkName.localizedCapitalized)_ChangeLog.md": """
+                    # \(frameworkName.localizedCapitalized) Updates
+                    
+                    @Metadata { @PageColor(yellow) }
+                    
+                    Learn about important changes to \(frameworkName.localizedCapitalized).
+                    
+                    ## Overview
+
+                    Browse notable changes in \(frameworkName.localizedCapitalized).
+                    
+                    ## Version: Diff between \(initialDocCArchiveVersion) and \(newerDocCArchiveVersion)
+
+                    
+                    ### Change Log
+                    
+                    #### Additions
+                    _New symbols added in \(newerDocCArchiveVersion) that did not previously exist in \(initialDocCArchiveVersion)._
+                                        
+                    \(additionLinks)
+                    
+                    
+                    #### Removals
+                    _Old symbols that existed in \(initialDocCArchiveVersion) that no longer exist in \(newerDocCArchiveVersion)._
+                                        
+                    \(removalLinks)
+                    
+                    """
+            ]
+        }
```

Comment:
Would be possible to add this template in `CatalogTemplateKind.swift` alongside the other templates we have in DocC? To me it seems like a better place to declare it.